src/erlcloud_sns.erl

```diff
@@ -571,6 +598,9 @@ extract_attribute(Nodes) ->
       erlcloud_xml:get_text("value", Node)}
      || Node <- Nodes].
 
+extract_topic_arn(Nodes) ->
+    [erlcloud_xml:decode([{arn, "TopicArn", text}], Node) || Node <- Nodes].
```

Comment:
Maybe that's better?

```
extract_topic_arn(Nodes) when is_list(Nodes) ->
    [extract_topic_arn(Node) || Node <- Nodes];
extract_topic_arn(Node) ->
    {arn, erlcloud_xml:get_text("TopicArn", Node)}.
```

Just to clearly separate a "list" and a "single element" cases. But that's a really minor thing, it's up to you to stay with your current solution or pick up this one